mongo/readpref/readpref.go

```diff
@@ -12,94 +12,145 @@ import (
 	"errors"
 	"fmt"
 	"time"
-
-	"go.mongodb.org/mongo-driver/v2/tag"
 )
 
-var (
-	errInvalidReadPreference = errors.New("can not specify tags, max staleness, or hedge with mode primary")
-)
+var errInvalidReadPreference = errors.New("can not specify tags, max staleness, or hedge with mode primary")
+
+// ReadPref determines which servers are considered suitable for read operations.
+type ReadPref struct {
+	Mode Mode
+
+	maxStaleness *time.Duration
```

Comment:
The current pattern prevents users from creating invalid read preferences, for example:
```
ms := 1 * time.Second
rp := readpref.ReadPref{
    Mode: readpref.PrimaryMode, 
    MaxStaleness: &ms, 
}
```

Concerning [server selection](https://github.com/mongodb/specifications/blob/82be6f2203ee35a6c40af36e5233b632ef402058/source/server-selection/server-selection.md#document-structure):

If the mode field is “primary”, the tags, maxStalenessSeconds, and hedge fields MUST be absent.

Concerning [max staleness](https://github.com/mongodb/specifications/blob/82be6f2203ee35a6c40af36e5233b632ef402058/source/max-staleness/max-staleness.md#api)

A connection string combining a positive maxStalenessSeconds with read preference mode “primary” MUST be considered invalid; this includes connection strings with no explicit read preference mode.

The only driver I've investigated that doesn't follow the pattern of validating in the read preference constructor is C (compared to [Node](https://github.com/mongodb/node-mongodb-native/blob/357ca086467bcfc425bcdba9844bcddcade670b0/src/read_preference.ts#L120), [Python](https://github.com/mongodb/mongo-python-driver/blob/3e5387e0ac77131c00b0912367e2237225d18d76/pymongo/read_preferences.py#L515), [PHP](https://github.com/mongodb/mongo-php-driver/blob/d3f3d7bc5619e2fa131626017d4edbadda6d2655/src/MongoDB/ReadPreference.c#L96)). C uses a [_mongoc_read_prefs_validate](https://github.com/mongodb/mongo-c-driver/blob/ef17740c31eb82655ac767a8d64f9e7e3a982180/src/libmongoc/src/mongoc/mongoc-read-prefs.c#L145) function to validate this behavior.